Realm/RLMProperty.mm

```diff
@@ -343,6 +343,9 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
     }
     else if ([_objcRawType isEqualToString:@"@\"NSNumber\""]) {
         const char *numberType = [propertyValue objCType];
+        if (!numberType) {
+            @throw RLMException(@"Can't persist NSNumber directly: use a Swift-native number type.");
```

Comment:
Are you suggesting we allow `NSNumber` to be used in Swift? That seems like it would provide confusing semantics since it user code doesn't annotate what type of number it stores.